File: php/features.php
<?php
declare(strict_types=1);

// InvoiceExtraction SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class InvoiceExtractionFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new InvoiceExtractionBaseFeature();
            case "test":
                return new InvoiceExtractionTestFeature();
            default:
                return new InvoiceExtractionBaseFeature();
        }
    }
}

File: py/features.py
# InvoiceExtraction SDK feature factory

from feature.base_feature import InvoiceExtractionBaseFeature
from feature.test_feature import InvoiceExtractionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: InvoiceExtractionBaseFeature(),
        "test": lambda: InvoiceExtractionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
